ui/server/src/main/scala/pl/touk/nussknacker/ui/api/ProcessAttachmentService.scala

```diff
@@ -1,57 +1,53 @@
 package pl.touk.nussknacker.ui.api
 
-import java.io.File
-import akka.stream.Materializer
-import akka.stream.scaladsl.{FileIO, Source}
+import akka.stream.{Materializer, StreamLimitReachedException}
+import akka.stream.scaladsl.{Source, StreamConverters}
 import akka.util.ByteString
 import com.typesafe.scalalogging.LazyLogging
 import pl.touk.nussknacker.engine.api.process.{ProcessId, VersionId}
-import pl.touk.nussknacker.ui.api.ProcessAttachmentService.AttachmentToAdd
-import pl.touk.nussknacker.ui.db.entity.AttachmentEntityData
+import pl.touk.nussknacker.ui.api.ProcessAttachmentService.{AttachmentDataWithName, AttachmentToAdd}
+import pl.touk.nussknacker.ui.config.AttachmentsConfig
 import pl.touk.nussknacker.ui.process.repository.ProcessActivityRepository
 import pl.touk.nussknacker.ui.security.api.LoggedUser
 import pl.touk.nussknacker.ui.util.CatsSyntax
 
+import java.io.IOException
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.control.NonFatal
+import scala.util.Using
 
-class ProcessAttachmentService(attachmentsBasePath: String, processActivityRepository: ProcessActivityRepository) extends LazyLogging {
+class ProcessAttachmentService(config: AttachmentsConfig, processActivityRepository: ProcessActivityRepository) extends LazyLogging {
 
   def saveAttachment(processId: ProcessId, processVersionId: VersionId, originalFileName: String, byteSource: Source[ByteString, Any])
                     (implicit ec: ExecutionContext, loggedUser: LoggedUser, mat: Materializer): Future[Unit] = {
-    val relativeFilePath = s"${processId.value}/${s"${System.currentTimeMillis()}-$originalFileName"}"
-    val attachmentFile = getAttachmentFile(relativeFilePath)
-    attachmentFile.getParentFile.mkdirs()
-    val fileSink = FileIO.toPath(attachmentFile.toPath)
-    byteSource.runWith(fileSink).flatMap { _ =>
-      val attachmentToAdd = AttachmentToAdd(processId, processVersionId, originalFileName, relativeFilePath)
-      processActivityRepository.addAttachment(attachmentToAdd).recoverWith { case NonFatal(ex) =>
-        logger.warn(s"Failure during writing attachment to db. Removing file ${attachmentFile}", ex)
-        attachmentFile.delete()
-        Future.failed(ex)
-      }
+    Future {
+      byteSource
+        .limitWeighted(config.maxSizeInBytes)(_.size)
+        .runWith(StreamConverters.asInputStream())
     }
+      .map(Using.resource(_)(_.readAllBytes()))
+      .recoverWith {
+        case e: IOException if e.getCause.isInstanceOf[StreamLimitReachedException] =>
+          Future.failed(new IllegalArgumentException(s"Maximum (${config.maxSizeInBytes} bytes) attachment size exceeded."))
```

Comment:
lets change FE accordingly so that user knows what happened, currently there's generic errror